sql/core/src/main/scala/org/apache/spark/sql/Dataset.scala

```diff
@@ -245,13 +262,17 @@ class Dataset[T] private[sql](
   }
 
   @transient private[sql] val logicalPlan: LogicalPlan = {
-    val plan = queryExecution.commandExecuted
-    if (sparkSession.sessionState.conf.getConf(SQLConf.FAIL_AMBIGUOUS_SELF_JOIN_ENABLED)) {
-      val dsIds = plan.getTagValue(Dataset.DATASET_ID_TAG).getOrElse(new HashSet[Long])
-      dsIds.add(id)
-      plan.setTagValue(Dataset.DATASET_ID_TAG, dsIds)
+    if (queryExecution.isLazyAnalysis) {
+      queryExecution.logical
```

Comment:
cc @cloud-fan

Reply:
What happens if the Dataset is a command?

Reply:
you can't put a command inside a subquery expression (neither in SQL).